File: native/macos/Sources/ModBuilderBWMac/Models/BuildRequest.swift
import Foundation

enum Region: String, Codable, CaseIterable, Identifiable {
    case EU
    case NA
    case RU
    case CUSTOM
    case NONE

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .NONE:
            return "Auto Detection"
        default:
            return rawValue
        }
    }

    var installerToken: String {
        switch self {
        case .NONE:
            return "AUTO"
        default:
            return rawValue
        }
    }
}

struct BuildRequest: Codable {
    var sources: [String]
    var outputDirectory: String
    var region: Region
    var gameRoot: String
    var modsFolderName: String
    var versionFolder: String
    var installerName: String
    var setupWindowTitle: String
    var installerIconPath: String
    var createZip: Bool
    var createInstallerMsi: Bool
}

File: native/windows/ModBuilderBW.Windows/Models/BuildRequest.cs
namespace ModBuilderBW.Windows.Models;

public enum Region
{
    EU,
    NA,
    RU,
    CUSTOM,
    NONE
}

public static class RegionExtensions
{
    public static string DisplayName(this Region region) => region switch
    {
        Region.NONE => "Auto Detection",
        _ => region.ToString()
    };

    public static string InstallerToken(this Region region) => region switch
    {
        Region.NONE => "AUTO",
        _ => region.ToString()
    };
}

public sealed class BuildRequest
{
    public required List<string> Sources { get; set; }
    public required string OutputDirectory { get; set; }
    public required Region Region { get; set; }
    public string GameRoot { get; set; } = string.Empty;
    public required string ModsFolderName { get; set; }
    public required string VersionFolder { get; set; }
    public string InstallerName { get; set; } = string.Empty;
    public string SetupWindowTitle { get; set; } = string.Empty;
    public string InstallerIconPath { get; set; } = string.Empty;
    public bool CreateZip { get; set; }
    public bool CreateInstallerMsi { get; set; }
}
